src/modules/minigame/common/floating_text/floating_text_util.gd

```diff
@@ -6,6 +6,9 @@ const FLOATING_TEXT_SCENE = preload("res://modules/minigame/common/floating_text
 
 ## Spawns a floating text instance at a given position with the specified value, then fades away.
 static func floating_text(position: Vector2, value: String, parent: Node) -> void:
+	if parent.get_tree().root == parent:
+		assert(false)  # don't add it to the root, causes lag & other problems
```

Comment:
Why does it cause lag?

Reply:
I'm not sure. panku seems to print something when this happens, and so it just gets worse and worse over time. in release (panku disabled) it's okay. I don't know enough about godot to say if adding children to the root is normal or problematic in general, but at least here with panku it is 🥹